### ISSUE TITLE
[BUG] Major Isar upgrade causing compilation errors due to incorrect versioning

### ISSUE THREAD

**Issue by HappyMakadiyaS** [JaffaKetchup/flutter_map_tile_caching#117]
### What is the bug?

Isar has been upgraded with Braking changes due to that FMTC throwing error while running the application.
```
../../../.pub-cache/hosted/pub.dev/flutter_map_tile_caching-7.2.0/lib/src/bulk_download/bulk_tile_writer.dart:134:27: Error: Required named parameter 'directory' must be provided.
  final db = Isar.openSync(
                          ^
../../../.pub-cache/hosted/pub.dev/isar-3.1.0/lib/src/isar.dart:118:15: Context: Found this candidate, but the arguments don't match.
  static Isar openSync(
              ^^^^^^^^
../../../.pub-cache/hosted/pub.dev/flutter_map_tile_caching-7.2.0/lib/src/bulk_download/bulk_tile_writer.dart:211:27: Error: Required named parameter 'directory' must be provided.
  final db = Isar.openSync(
                          ^
../../../.pub-cache/hosted/pub.dev/isar-3.1.0/lib/src/isar.dart:118:15: Context: Found this candidate, but the arguments don't match.
  static Isar openSync(
              ^^^^^^^^
../../../.pub-cache/hosted/pub.dev/flutter_map_tile_caching-7.2.0/lib/src/providers/image_provider.dart:251:27: Error: Required named parameter 'directory' must be provided.
  final db = Isar.openSync(
                          ^
../../../.pub-cache/hosted/pub.dev/isar-3.1.0/lib/src/isar.dart:118:15: Context: Found this candidate, but the arguments don't match.
  static Isar openSync(
              ^^^^^^^^
../../../.pub-cache/hosted/pub.dev/flutter_map_tile_caching-7.2.0/lib/src/db/defs/store_descriptor.g.dart:16:33: Error: Constant evaluation error:
const DbStoreDescriptorSchema = CollectionSchema(
                                ^
../../../.pub-cache/hosted/pub.dev/isar-3.1.0/lib/src/schema/collection_schema.dart:24:24: Context: This assertion failed with message: Outdated generated code. Please re-run code generation using the latest generator.
          Isar.version == version,
                       ^
../../../.pub-cache/hosted/pub.dev/flutter_map_tile_caching-7.2.0/lib/src/db/defs/store_descriptor.g.dart:16:7: Context: While analyzing:
const DbStoreDescriptorSchema = CollectionSchema(
.....
....
```
Pub.dev has Analysis issue as well
<img width="699" alt="image" src="https://user-images.githubusercontent.com/97177197/232381004-64314bb9-785c-4f29-b698-7cfc479ea5ca.png">

**Could you please fix it on priority? This is a blocker for us for a production application we are working on.**

### Platforms Affected

Android, iOS

### Severity

Erroneous: Prevents normal functioning and causes errors in the console

### Frequency

Consistently: Always occurs at the same time and location

### Requirements

- [X] I agree to follow this project's [Code of Conduct](https://github.com/jaffaketchup/flutter_map_tile_caching/blob/main/CODE_OF_CONDUCT.md)
- [X] I have checked for known issues in the GitHub tracker and [on the docs site](https://fmtc.jaffaketchup.dev/known-issues-and-reporting#known-issues)
- [X] I have completed the pre-report checks listed [on the docs site](https://fmtc.jaffaketchup.dev/known-issues-and-reporting#reporting-a-bug)
- [X] If I am using FMTC in a proprietary project, I have read and understood [this important legal notice](https://fmtc.jaffaketchup.dev/#proprietary-licensing)

**Comment by JaffaKetchup** [comment]
@HappyMakadiyaS I have not yet claimed support for Isar v3.1. Unfortunately their change was breaking, but they did not follow semver. Now the package resolver forces usage of v3.1, even though FMTC is only compatible with v3.0.

As a workaround, please use dependency overrides to pin the Isar dependency to v3.0.6-dev.0.

**Comment by JaffaKetchup** [comment]
In addition, I have reason to believe you are breaking the terms of the GPL license. If your application is proprietary, please get in touch through email.
This also applies to Simform (cc @dhaval-k-simformsolutions).

**Comment by HappyMakadiyaS** [comment]
@JaffaKetchup 
Thanks for the quick reply. We are not supposed to break any licence agreement here. We are just checking the feasibility of this package with our use case, if its fits in our use case we surely reach out to you for licence purchase things. Hope my answers makes sense to you. Please let me know any correction needed at my end.

**Comment by sikandernoori** [comment]
@JaffaKetchup 
> As a workaround, please use dependency overrides to pin the Isar dependency to v3.0.6-dev.0.

```
Connecting to VM Service at ws://127.0.0.1:52112/hfwEzjDqjW4=/ws
[ +430 ms] [VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: IsarError: Could not initialize IsarCore library for processor architecture "ios_arm64". If you create a Flutter app, make sure to add isar_flutter_libs to your dependencies.
[        ] IsarError: Incorrect Isar Core version: Required 3.0.6-dev.0 found 3.1.0. Make sure to use the latest isar_flutter_libs. If you have a Dart only project, make sure that old Isar Core binaries are deleted.
[        ] #0      initializeCoreBinary
isar_core.dart:78
[        ] #1      openIsar
open.dart:81
[        ] #2      Isar.open
isar.dart:107
[        ] #3      FMTCRegistry.initialise
registry.dart:57
[        ] #4      FlutterMapTileCaching.initialise
fmtc.dart:102
[        ] <asynchronous suspension>
[        ] #5      initializeFMTC
main.dart:214
[        ] <asynchronous suspension>
[        ] #6      runTrackr
main.dart:170
[        ] <asynchronous suspension>
[        ] #7      main
main.dart:136
[        ] <asynchronous suspension>
[   +3 ms] flutter: [token:, 83356cd8c4af8b1351d1e9469e22872123af0858cdc2005990d6fe461f7f2034]
```

**Comment by JaffaKetchup** [comment]
Also pin all related packages to the same version, like 'isar_flutter_libs'.

**Comment by JaffaKetchup** [comment]
Fixed in the latest prerelease: https://pub.dev/packages/flutter_map_tile_caching/versions/8.0.0-dev.5.
I'd now recommend using the prerelease over the non-prerelease.

**Comment by JaffaKetchup** [comment]
**See the Important Temporary Notice posted on the front page of https://fmtc.jaffaketchup.dev/ for more information about this issue and its workarounds.**

![image](https://user-images.githubusercontent.com/58115698/232900451-86943964-6dec-4c56-8c14-eb2468743c16.png)

**Comment by JaffaKetchup** [comment]
This has been fixed in #114. Thanks for all your perseverance!
